### DISCUSSION TITLE
Custom Bootstrapper with Transparent wpf window

### DISCUSSION THREAD

**Discussion by j-hn** [wixtoolset/issues#7719] [Questions] [answered]
I had a custom boostrapper with a WPF UI in wix 3.14. which was ok.
I converted this to wix 4, and eventually got it running, (following the examples)
however.... my main wpf window has AllowsTransparency="true" this causes the install to fail, If I set it to false it then installs fine.
it worked fine in wix 3.x.
any way I can get may transparency back?

Thanks

Inner Exception 1:
InvalidOperationException: Cannot change AllowsTransparency after a Window has been shown or WindowInteropHelper.EnsureHandle has been called.

**Comment by nirbar** [comment, marked as answer]
I'm running a BA that has `AllowsTransparency="true"` without any error.
Did you make sure that you're not calling `EnsureHandle` before `InitializeComponent` in RootView c'tor?

**Reply by j-hn** [reply]
Oh Blimey! I'm an idiot!

thanks for your help
